src/diffusers/schedulers/scheduling_ddpm.py

```diff
@@ -214,16 +214,17 @@ def _get_variance(self, t, predicted_variance=None, variance_type=None):
         # and sample from it to get previous sample
         # x_{t-1} ~ N(pred_prev_sample, variance) == add variance to pred_sample
         variance = (1 - alpha_prod_t_prev) / (1 - alpha_prod_t) * current_beta_t
+        variance = torch.clamp(variance, min=1e-20)
```

Comment:
we always take the log of `variance` so we should clamp it once

Reply:
Numerical precisions galore :)

Reply:
https://github.com/huggingface/diffusers/pull/3006/commits/deaf1d614747e2112afab49d5d37bf363c611521